src/bms/parse/model.rs

```diff
@@ -200,12 +201,12 @@ pub struct Notes {
     pub wav_files: HashMap<ObjId, PathBuf>,
     // objects stored in obj is sorted, so it can be searched by bisection method
     /// BGM objects, indexed by time. #XXX01:ZZ... (BGM placement)
-    pub bgms: BTreeMap<ObjTime, Vec<ObjId>>,
-    /// All note objects, indexed by ObjId. #XXXYY:ZZ... (note placement)
-    pub objs: HashMap<ObjId, Vec<Obj>>,
+    pub bgms: MultiMap<ObjTime, ObjId>,
+    /// All note objects, indexed by ObjId (like BMSON's `sound_channel`). #XXXYY:ZZ... (note placement)
+    pub objs: HashMap<ObjId, MultiMap<ObjTime, Obj>>,
```

Comment:
`objs` is master data source of all note objects. To seek notes efficiently, I added `ids_by_key` abd `bgms`. Please use `MultiMap` for `objs` like this:
```rs
pub objs: MultiMap<ObjId, Obj>,
```